File: app/src/androidTest/java/com/leaguechampions/ui/champions/ChampionsActivityTest.java
package com.leaguechampions.ui.champions;

import android.support.test.InstrumentationRegistry;
import android.support.test.rule.ActivityTestRule;
import android.support.test.runner.AndroidJUnit4;

import com.leaguechampions.ui.championdetails.ChampionDetailsRobot;
import com.leaguechampions.ui.settings.SettingsRobot;
import com.leaguechampions.utils.PrefUtils;

import org.junit.After;
import org.junit.Before;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

@RunWith(AndroidJUnit4.class)
public class ChampionsActivityTest {

    @Rule
    public ActivityTestRule<ChampionsActivity> activityTestRule = new ActivityTestRule<ChampionsActivity>(ChampionsActivity.class) {
        @Override
        protected void beforeActivityLaunched() {
            PrefUtils.setMockMode(InstrumentationRegistry.getTargetContext(), true);
            super.beforeActivityLaunched();
        }
    };

    @Before
    public void setUp() {

    }

    @After
    public void tearDown() {

    }

    @Test
    public void displayChampionsList() {
        new ChampionsRobot()
                .verifyChampionsAreDisplayed();
    }

    @Test
    public void displayChampionsList_ClickChampion_DisplayChampionDetails() {
        new ChampionsRobot()
                .clickChampionAtPosition(1);

        new ChampionDetailsRobot()
                .verifyChampionIsDisplayed("Riven");
    }

    @Test
    public void showSettings() {
        new ChampionsRobot()
                .openMenu()
                .clickMenuSettings();

        new SettingsRobot()
                .checkVersionIsDisplayed();
    }
}

File: app/src/androidTest/java/com/leaguechampions/ui/champions/ChampionsActivityTest.kt
package com.leaguechampions.ui.champions

import androidx.test.InstrumentationRegistry
import androidx.test.rule.ActivityTestRule
import androidx.test.runner.AndroidJUnit4
import com.leaguechampions.features.champions.champions.champions.ChampionsActivity

import com.leaguechampions.ui.championdetails.ChampionDetailsRobot
import com.leaguechampions.ui.settings.SettingsRobot
import com.leaguechampions.core.utils.PrefUtils

import org.junit.After
import org.junit.Before
import org.junit.Rule
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
class ChampionsActivityTest {

    @Rule
    @JvmField
    var activityTestRule: ActivityTestRule<ChampionsActivity> = object : ActivityTestRule<ChampionsActivity>(ChampionsActivity::class.java) {
        override fun beforeActivityLaunched() {
            com.leaguechampions.core.utils.PrefUtils.setMockMode(InstrumentationRegistry.getTargetContext(), true)
            super.beforeActivityLaunched()
        }
    }

    @Before
    fun setUp() {

    }

    @After
    fun tearDown() {

    }

    @Test
    fun displayChampionsList() {
        ChampionsRobot()
                .verifyChampionsAreDisplayed()
    }

    @Test
    fun displayChampionsList_ClickChampion_DisplayChampionDetails() {
        ChampionsRobot()
                .clickChampionAtPosition(1)

        ChampionDetailsRobot()
                .verifyChampionIsDisplayed("Riven")
    }

    @Test
    fun showSettings() {
        ChampionsRobot()
                .openMenu()
                .clickMenuSettings()

        SettingsRobot()
                .checkVersionIsDisplayed()
    }
}
